### DISCUSSION TITLE
How to just enable https next to http for admin interface

### DISCUSSION THREAD

**Discussion by sanderlv** [AdguardTeam/AdGuardHome#4956] [Q&A]
Just very simple question. Can adguard be enabled to be accessed via https and a self signed certificate?

**Comment by ghost** [comment]
Yes, look in Encryption settings.

**Reply by sanderlv** [reply]
Ofcourse did that. How to do it? What setting?

**Comment by ghost** [comment]
Tick "Enable Encryption" and tick "Redirect to HTTPS automatically". The Certificate part is on the bottom of the same section. You can either copy/paste certificate and key or provide a directory for each.

Unfortunately, by using settings above you also enable your server to allow for DoH requests using the same certificate-key pair you set for administrator login. I personally don't like that and would want to have separate certificate-key pairs or have options to indepently assigns allowance for login and DoH encryption. That may happen (or not) in the future.

**Reply by sanderlv** [reply]
Hi, thank you for replying. So to be sure, there is no option such as like just enbale https. But instead manual certificates need to be copy pasted. Not sure about that, I never have created a certificate like that. Is there no option to have some for of self signed cert?

I run the adguard only internally and it is not reachable via internet. How can I create a cert this way?

**Comment by ghost** [comment]
Yes, you need to generate a certificate-key pair and then paste them individually into correct fields in AdGuard Home. To generate such keys in Windows OS, you can use binaries package (1.1.1q) from here - https://kb.firedaemon.com/support/solutions/articles/4000121705-openssl-3-0-and-1-1-1-binary-distributions-for-microsoft-windows . The needed commands are listed on that site. Its quite simple. 

Once you generate certificate-key pair, paste each one into corresponding field in AdGuard Home.

**Comment by sanderlv** [comment]
Hi, it's not so suite simple as you might think it is for me...

![image](https://user-images.githubusercontent.com/16528916/193883436-855f2f39-75b3-4184-8e47-378f5660197a.png)

Cant save
